sips/sip-025/sip-025-sbtc.md

```diff
@@ -0,0 +1,906 @@
+
+![Nakamoto Flow Diagrams (1)](https://github.com/stacksgov/sips/assets/459947/e8da8826-1303-4891-83fb-a568ec9d3f9d)
+
+_Figure 2: Protocol diagram for a successful peg-out request.  The user submits
+a peg-out request to Stacks, which locks their requested sBTC tokens until the
+start of the next reward cycle.  If the peg-out succeeds, then they receive BTC from
+the Stackers (minus a BTC transaction fee), and their sBTC gets burnt._
+
+When Stackers see the peg-out request, and if they are able to act on it (i.e.
+the wallet has enough UTXOs free), they execute a WSTS signing round to
+generate a Bitcoin transaction from the peg wallet's UTXOs which pays out to the
+user's requested address.  Because all pegged-in BTC is spendable by the current
+set of Stackers via an aggregate Schnorr signature, Stackers simply create the
+scriptSigs for sufficiently many UTXOs to fund a fulfillment UTXO for the user.
+
+Because fulfilling the redemption request occurs through WSTS, there is a
+possibility of a quorum failure.  As such, the Stackers may instead vote to
+abstain or even to refuse the peg-out request.  This is explained in the
+"Failure Modes" section.
+
+The BTC redemption transaction contains the following fields:
+
+* An `OP_RETURN` identifying the peg-out request in the `.sbtc` contract
+
+* A new UTXO that redeems the user's sBTC for BTC
+
+* A change UTXO that consolidates the remaining unused BTC from the consumed
+  wallet UTXOs
+
+Once the Stackers have created the BTC redemption transaction, they both send it
+to the Bitcoin peer, and send a Stacks transaction with a copy of the Bitcoin
+transaction to `.sbtc`, so that the transaction is _also_ stored publicly in the
+`.sbtc` contract's data space.  This is done for failure recovery: if the
+Bitcoin transaction does not confirm quickly, the Stackers can try again (see
+below).  Also, the existence of the signed Bitcoin transaction in the `.sbtc`
+data space serves as proof that the Stackers handled the peg-out request in a
+timely manner, even if it was not confirmed on Bitcoin as quickly as desired.
+When the `.sbtc` contract processes the copy of the BTC redemption transaction,
+it marks the UTXOs it consumes as "tentatively used" so that a
+concurrently-generated BTC redemption transaction dees not also use them.
+
+Once the BTC redemption transaction confirms on Bitcoin, a Stacks miner
+processes it as a Stacks-on-Bitcoin transaction.  The act of processing this
+transaction is the act of calling into the `.sbtc` contract to mark the pending
+sBTC peg-out as fulfilled.  This burns the user's locked sBTC, and marks the
+peg-out operation as complete.  It also updates the `.sbtc` transaction's
+knowledge of the wallet's UTXO set by clearing the UTXOs consumed by the
+fulfillment, and adding the change UTXO as a new UTXO that the wallet controls.
+
+### Peg-Out Congestion
+
+A key responsibility of the `.sbtc` wallet is to keep track of which Bitcoin
```

Comment:
[nit] what's the distinction between `.sbtc` wallet and "peg wallet"? If they're the same, considering using one variant for consistency

Reply:
`.sbtc` is a smart contract.  The peg wallet is the set of UTXOs that can be spent by the Stackers to process withdrawals.